models/DataObject/Traits/SimpleComparisonTrait.php

```diff
@@ -29,4 +29,22 @@ public function isEqual($oldValue, $newValue)
     {
         return $oldValue == $newValue;
     }
+
+    /**
+     * @param array|null $array1
+     * @param array|null $array2
+     *
+     * @return bool
+     */
+    protected function isEqualArray($array1, $array2)
+    {
+        $array1 = array_filter(is_array($array1) ? $array1 : []);
+        $array2 = array_filter(is_array($array2) ? $array2 : []);
+
+        if (count($array1) != count($array2)) {
+            return false;
+        }
+
+        return serialize($array1) === serialize($array2);
```

Comment:
serialize is not the fastet option, I'd prefer a loop actually comparing values. In that case maybe json_encode? still faster than serialize

Reply:
we are not sure about array structure here...it can have nested objects with protected properties.

Reply:
Why not `$array1 == $array2`? When there are objects as array items their attributes get compared.
```php
class A {
    protected $a;
    
    public function __construct($a) {
        $this->a = $a;
    }
}

$array1 = [new A(1), new A(2)];
$array2 = [new A(1), new A(2)];
var_dump($array1 == $array2); // returns true

$array1 = [new A(1), new A(2)];
$array2 = [new A(2), new A(2)];
var_dump($array1 == $array2); // returns false
```

Reply:
I'm afraid this evaluates to true, even if ordering is different so it is not correct for some data types where ordering is allowed, so something like: 

```php
$arr1 = [ 0 => $object1, 1 => $object2];
$arr2 = [ 1 => $object2, 0 => $object1];

var_dump($arr1 == $arr2); // returns true

```
but yes, I'll test some cases with identity operator `===`. thanks!

Reply:
you are right, it works with just `==` comparison. changed it ;)